common/lib/xmodule/xmodule/partitions/tests/test_partitions.py

```diff
@@ -325,9 +325,19 @@ def test_get_user_group_id_for_partition(self):
 
         # switch to the second group and verify that it is returned for the user
         self.user_partition.scheme.current_group = groups[1]    # pylint: disable=no-member
-        group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #self.assertEqual(group2_id, groups[1].id)    # pylint: disable=no-member
+
+        partition_service2 = StaticPartitionService(
+            [self.user_partition],
+            user=Mock(username='ma', email='ma@edx.org', is_staff=False, is_active=True),
+            course_id=self.course.id,
+            track_function=Mock()
+        )
+        group2_id = partition_service2.get_user_group_id_for_partition(user_partition_id)
```

Comment:
i thought at least in the case of CohortPartitionScheme that there were parameters to cancel the auto-assignment side effect in get group for user - maybe i'm misremembering?  Either way, in practice, i do think that within the lifecycle of a web request the possibility of a return value from either of the implemented PartitionSchemes changing is very close to zero.  It's only during other as-yet-unimagined, longer-running transactions or tasks where i think mandatory caching could become confusing or limiting.